presto-hive-functions/src/main/java/com/facebook/presto/hive/functions/gen/ScalarMethodHandles.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.hive.functions.gen;
+
+import com.facebook.presto.bytecode.ClassDefinition;
+import com.facebook.presto.bytecode.MethodDefinition;
+import com.facebook.presto.bytecode.Parameter;
+import com.facebook.presto.bytecode.ParameterizedType;
+import com.facebook.presto.bytecode.expression.BytecodeExpression;
+import com.facebook.presto.common.predicate.Primitives;
+import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.common.type.TypeSignature;
+import com.facebook.presto.hive.functions.scalar.ScalarFunctionInvoker;
+import com.facebook.presto.spi.function.Signature;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.bytecode.Access.FINAL;
+import static com.facebook.presto.bytecode.Access.PRIVATE;
+import static com.facebook.presto.bytecode.Access.PUBLIC;
+import static com.facebook.presto.bytecode.Access.STATIC;
+import static com.facebook.presto.bytecode.Access.a;
+import static com.facebook.presto.bytecode.Parameter.arg;
+import static com.facebook.presto.bytecode.ParameterizedType.type;
+import static com.facebook.presto.bytecode.expression.BytecodeExpressions.newArray;
+import static com.facebook.presto.common.block.MethodHandleUtil.methodHandle;
+import static com.facebook.presto.hive.functions.gen.CompilerUtils.defineClass;
+import static com.facebook.presto.hive.functions.gen.CompilerUtils.makeClassName;
+
+public final class ScalarMethodHandles
+{
+    private static final String CLASS_NAME = "HiveScalarFunction";
+    private static final String METHOD_NAME = "callEvaluate";
+
+    private ScalarMethodHandles()
+    {
+    }
+
+    /**
+     * Generates an unbound MethodHandle by {@code signature}.
+     * <p>
+     * For example, if {@code signature} describes a method like
+     * <pre>
+     * {@code
+     * IntegerType func(
+     *              IntegerType i,
+     *              VarcharType j)
+     * }
+     * </pre>,
+     * a method is generated as
+     * <pre>
+     * {@code
+     * static Long callEvaluate(
+     *              ScalarFunctionInvoker invoker,
+     *              long input_0,
```

Comment:
`Long`?

Reply:
Long is used for most integer types in presto, right?

Reply:
I mean it should be uppercase `Long` rather than lowercase primitive type `long`.